scheduler/scheduler.go

```diff
@@ -495,19 +500,24 @@ func (s *scheduler) startTask(id, source string) []serror.SnapError {
 		if err != nil {
 			errs = append(errs, serror.New(err))
 		} else {
-			errs = mgr.SubscribeDeps(t.ID(), depGroupMap[k].Metrics, cps)
+			// it is the only place where requested metrics are non-empty (t.workflow.metrics)
+			// on LoadPluginEvent SubscribeDeps will be called with empty list of requested metrics
+			// it is the only place where resulted metrics is empty (on Load/Unload expansion will be done before
+			// SubscribeDeps being called
+			errs = mgr.SubscribeDeps(t.ID(), []core.Metric{}, t.workflow.metrics, cps, t.workflow.configTree)
```

Comment:
We should consider making the exported `SubscribeDeps` take only requested metrics.  

I'm going to send you some more details on this.